src/app/form-builder/form-builder.component.ts

```diff
@@ -0,0 +1,191 @@
+  }
+
+  getOptions(id: number): any {
+    return this.listStyles.find(el => el.id === id).params['options'];
+  }
+
+  getExampleListStyle(id: number): any {
+    return this.listStylesRow.find(el => el.id === id).styles;
+  }
+
+  getRowID(order: any): number {
+    return this.listStylesRow[order].id;
+  }
+
+  addRow(event: any): void {
+    this.basket.push([]);
+    this.rowStore$.dispatch(new SetNewRowAction({ id: this.counterRowID }));
+    this.counterRowID++;
+  }
+
+  deleteRow(event: any): void {
+    let idRow = Number(document.querySelector('.activeRow').id);
+    if (this.basket.length > 1) {
+      let rowContainer = document.querySelector('.activeRow');
+      if (rowContainer.children.length === 0) {
+        this.basket.splice(idRow, 1);
+        this.rowStore$.dispatch(new DeleteRowAction({ id: idRow }))
+        this.deleteRowBtnStatus = true;
+      } else {
+        document.querySelector('.activeRow').classList.remove("activeRow");
+        this.deleteRowBtnStatus = true;
+        alert("Row are not empty! Delete Elements in row and after delete row!")
+      }
+    } else {
+      document.querySelector('.activeRow').classList.remove("activeRow");
+      this.deleteRowBtnStatus = true;
+    }
+  }
+
+  setActiveRow(event: any): void {
+    if (event.target.classList.contains("activeRow") && event.target.classList.contains("example-list")) {
+      event.target.classList.toggle("activeRow");
+      this.deleteRowBtnStatus = true;
+    } else if (!event.target.classList.contains("activeRow") && event.target.classList.contains("example-list")) {
+      document.querySelectorAll('.activeRow').forEach(elem => elem.classList.remove("activeRow"));
+      event.target.classList.toggle("activeRow");
+      this.deleteRowBtnStatus = false;
+      this.targetRow.emit(event.target);
+    }
+  }
+
+  deleteElem(event: any): void {
+    let containerElem = document.querySelector('.activeElemForm');
+    let elem = document.querySelector('.activeElemForm .elementForm');
+    let elemID = Number(elem.getAttribute("data-id"));
+
+    containerElem.remove();
+    for (let key in this.basket) {
+      this.basket[key] = this.basket[key].filter(el => el.id !== elemID)
+    }
+    this.store$.dispatch(new DeleteElemAction({ id: elemID }));
+  }
+
+  drop(event: CdkDragDrop<any[]>) {
+    let containerID = Number(event.container.element.nativeElement.dataset.position);
+    if (event.previousContainer === event.container) {
+      moveItemInArray(this.basket[containerID], event.previousIndex, event.currentIndex);
+    } else {
+
+      if (this.basket[containerID].length < 4) {
+        let data = { character: event.previousContainer.data[event.previousIndex].character, id: this.counterID };
+        this.basket[containerID].push(data);
+        this.store$.dispatch(new SetNewElemAction({ type: data.character, id: this.counterID }))
+        this.counterID++;
+      } else {
+        alert("Maximum 4 elements in row!")
+      }
+    }
+  }
```

Comment:
find a better solution
```suggestion
  drop(event: CdkDragDrop<any[]>): void {
    let containerID = Number(event.container.element.nativeElement.dataset.position);
    if (event.previousContainer === event.container) {
      moveItemInArray(this.basket[containerID], event.previousIndex, event.currentIndex);
    } else {

      if (this.basket[containerID].length < 4) {
        let data = { character: event.previousContainer.data[event.previousIndex].character, id: this.counterID };
        this.basket[containerID].push(data);
        this.store$.dispatch(new SetNewElemAction({ type: data.character, id: this.counterID }))
        this.counterID++;
      } else {
        alert("Maximum 4 elements in row!")
      }
    }
  }
```